[Bug]: Regression in widget behavior

As of https://github.com/matplotlib/matplotlib/pull/29855 (according to `git bisect`) on nightly wheels in MNE-Python we [now see](https://github.com/mne-tools/mne-python/actions/runs/17794760025/job/50579948170?pr=13423#step:19:6189):
```
_____________________ test_plot_raw_selection[matplotlib] ______________________
mne/viz/tests/test_raw.py:418: in test_plot_raw_selection
    fig._fake_keypress(sel_fig.mne.close_key, fig=sel_fig)
mne/viz/_mpl_figure.py:2249: in _fake_keypress
    _fake_keypress(fig, key)
mne/viz/utils.py:813: in _fake_keypress
    fig.canvas.callbacks.process(
/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/matplotlib/cbook.py:365: in process
    func(*args, **kwargs)
/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/matplotlib/widgets.py:2338: in on_key_press
    self.clear()
/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/matplotlib/widgets.py:2383: in clear
    self.update()
/opt/hostedtoolcache/Python/3.13.7/x64/lib/python3.13/site-packages/matplotlib/widgets.py:2240: in update
    self.canvas.restore_region(self.background)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^
E   AttributeError: 'FigureCanvasBase' object has no attribute 'restore_region'
```
Locally I can see:
```
../matplotlib/lib/matplotlib/widgets.py:2221: in update
    self.canvas.restore_region(self.background)
E   AttributeError: 'FigureCanvasBase' object has no attribute 'restore_region'
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/larsoner/python/matplotlib/lib/matplotlib/widgets.py(2221)update()
-> self.canvas.restore_region(self.background)
(Pdb) p type(self)
<class 'matplotlib.widgets.LassoSelector'>
(Pdb) p self.canvas
<matplotlib.backend_bases.FigureCanvasBase object at 0x7f47ede5ca50>
(Pdb) u
(Pdb) u
(Pdb) u
(Pdb) u
> /home/larsoner/python/mne-python/mne/viz/utils.py(813)_fake_keypress()
-> fig.canvas.callbacks.process(
(Pdb) p fig
<MNESelectionFigure size 300x700 with 3 Axes>
```
[`MNESelectionFigure`](https://github.com/mne-tools/mne-python/blob/662fb510097e38667a342e41fa525330154c5f74/mne/viz/_mpl_figure.py#L275) subclasses [`MNEFigure`](https://github.com/mne-tools/mne-python/blob/662fb510097e38667a342e41fa525330154c5f74/mne/viz/_mpl_figure.py#L85) which subclasses `matplotlib.figure.Figure`.

I'm hoping this is enough to see the problem and know the solution. But if not, let me know and I'll take the time to whittle this down to a MWE! There might also be some way we are misusing/simulating keypresses in tests or something, but we haven't changed that code in years AFAIK so if this breaks things for us I wouldn't be surprised if it breaks things for others, too.

@timhoffm I still experience a variant of this on latest `main` on https://github.com/mne-tools/mne-python/pull/13569 which removes our workaround:
```
$ pytest mne/viz/tests/test_raw.py -k test_plot_raw_selection[matplotlib]
```
gives:

<details>
<summary>Traceback</summary>

```
=============================================================================== test session starts ================================================================================
platform linux -- Python 3.13.7, pytest-8.3.5, pluggy-1.6.0
PySide6 6.10.1 -- Qt runtime 6.10.1 -- Qt compiled 6.10.1
MNE 1.12.0.dev16+g825652aa74 -- /home/larsoner/python/mne-python/mne
rootdir: /home/larsoner/python/mne-python
configfile: pyproject.toml
plugins: dependency-0.6.0, anyio-4.9.0, cov-6.1.1, flaky-3.8.1, qt-4.4.0, rerunfailures-16.1, timeout-2.4.0
collected 71 items / 70 deselected / 1 selected                                                                                                                                    

mne/viz/tests/test_raw.py F                                                                                                                                                  [100%]



===================================================================================== FAILURES =====================================================================================
_______________________________________________________________________ test_plot_raw_selection[matplotlib] ________________________________________________________________________
mne/viz/tests/test_raw.py:419: in test_plot_raw_selection
    fig._close_event(sel_fig)
mne/viz/_mpl_figure.py:2250: in _close_event
    _close_event(fig)
mne/fixes.py:658: in _close_event
    fig.canvas.callbacks.process(
../matplotlib/lib/matplotlib/cbook.py:365: in process
    func(*args, **kwargs)
mne/viz/utils.py:1193: in _close_event
    fig.lasso.disconnect()
mne/viz/utils.py:1780: in disconnect
    self.canvas.draw_idle()
../matplotlib/lib/matplotlib/backend_bases.py:1981: in draw_idle
    self.draw(*args, **kwargs)
../matplotlib/lib/matplotlib/backends/backend_agg.py:383: in draw
    self.figure.draw(self.renderer)
../matplotlib/lib/matplotlib/artist.py:94: in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
../matplotlib/lib/matplotlib/artist.py:71: in draw_wrapper
    return draw(artist, renderer)
../matplotlib/lib/matplotlib/figure.py:3275: in draw
    DrawEvent("draw_event", self.canvas, renderer)._process()
../matplotlib/lib/matplotlib/backend_bases.py:1194: in _process
    self.canvas.callbacks.process(self.name, self)
../matplotlib/lib/matplotlib/cbook.py:365: in process
    func(*args, **kwargs)
../matplotlib/lib/matplotlib/widgets.py:1746: in _clear
    self._save_blit_background(self.canvas.copy_from_bbox(self.ax.bbox))
E   AttributeError: 'FigureCanvasBase' object has no attribute 'copy_from_bbox'
```

</details>

Maybe this one should be reopened?

--- a/lib/matplotlib/widgets.py
+++ b/lib/matplotlib/widgets.py
@@ -1108,7 +1108,7 @@ class CheckButtons(AxesWidget):
         if actives is None:
             actives = [False] * len(labels)
 
-        self._useblit = useblit and self.canvas.supports_blit  # TODO: make dynamic
+        self._useblit = useblit
 
         ys = np.linspace(1, 0, len(labels)+2)[1:-1]
 
@@ -1136,7 +1136,10 @@ class CheckButtons(AxesWidget):
             **cbook.normalize_kwargs(check_props, collections.PathCollection),
             'marker': 'x',
             'transform': ax.transAxes,
-            'animated': self._useblit,
+            'animated': self._useblit and self.canvas.supports_blit,
+            # TODO: This may need an update when switching out the canvas.
+            #       Can set this to `_useblit` only and live with the animated=True
+            #       overhead on unsupported backends.
         }
         check_props.setdefault('facecolor', check_props.pop('color', 'black'))
         self._checks = ax.scatter([0.15] * len(ys), ys, **check_props)
@@ -1155,7 +1158,8 @@ class CheckButtons(AxesWidget):
         """Internal event handler to clear the buttons."""
         if self.ignore(event) or self.canvas.is_saving():
             return
-        self._save_blit_background(self.canvas.copy_from_bbox(self.ax.bbox))
+        if self._useblit and self.canvas.supports_blit:
+            self._save_blit_background(self.canvas.copy_from_bbox(self.ax.bbox))
         self.ax.draw_artist(self._checks)
 
     @_call_with_reparented_event
@@ -1260,7 +1264,7 @@ class CheckButtons(AxesWidget):
         self._checks.set_facecolor(facecolors)
 
         if self.drawon:
-            if self._useblit:
+            if self._useblit and self.canvas.supports_blit:
                 background = self._load_blit_background()
                 if background is not None:
                     self.canvas.restore_region(background)
@@ -1701,7 +1705,7 @@ class RadioButtons(AxesWidget):
 
         ys = np.linspace(1, 0, len(labels) + 2)[1:-1]
 
-        self._useblit = useblit and self.canvas.supports_blit  # TODO: make dynamic
+        self._useblit = useblit
 
         label_props = _expand_text_props(label_props)
         self.labels = [
@@ -1716,7 +1720,11 @@ class RadioButtons(AxesWidget):
             **radio_props,
             'marker': 'o',
             'transform': ax.transAxes,
-            'animated': self._useblit,
+            'animated': self._useblit and self.canvas.supports_blit,
+            # TODO: This may need an update when switching out the canvas.
+            #       Can set this to `_useblit` only and live with the animated=True
+            #       overhead on unsupported backends.
+
         }
         radio_props.setdefault('edgecolor', radio_props.get('color', 'black'))
         radio_props.setdefault('facecolor',
@@ -1743,7 +1751,8 @@ class RadioButtons(AxesWidget):
         """Internal event handler to clear the buttons."""
         if self.ignore(event) or self.canvas.is_saving():
             return
-        self._save_blit_background(self.canvas.copy_from_bbox(self.ax.bbox))
+        if self._useblit and self.canvas.supports_blit:
+            self._save_blit_background(self.canvas.copy_from_bbox(self.ax.bbox))
         self.ax.draw_artist(self._buttons)
 
     @_call_with_reparented_event
@@ -1836,7 +1845,7 @@ class RadioButtons(AxesWidget):
         self._buttons.set_facecolor(button_facecolors)
 
         if self.drawon:
-            if self._useblit:
+            if self._useblit and self.canvas.supports_blit:
                 background = self._load_blit_background()
                 if background is not None:
                     self.canvas.restore_region(background)

